### PR TITLE
SAA-1357 add missing call to monitoring service where a subsequent job can fail (not run) due to the first not working.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Adding (missed) call to monitoring service in the safe job runner.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ManageAttendancesService.kt [hunk 1]
```diff
@@ -83,6 +83,8 @@ class ManageAttendancesService(
                 log.info("Sending sync event for attendance ID ${saved.attendanceId} ${saved.prisonerNumber} ${saved.scheduledInstance.activitySchedule.description}")
                 outboundEventsService.send(OutboundEvent.PRISONER_ATTENDANCE_CREATED, saved.attendanceId)
               }
+            }.onFailure {
+              log.error("Error occurred saving attendances", it)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ManageAttendancesService.kt:87]: I noticed we are potentially swallowing errors here without even knowing about it.

Given we effectively have a try catch here I did wonder if we should also be raising it via the monitoring service or do away with the try catch and let it propagate up the stack?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/job/SafeJobRunner.kt [hunk 2]
```diff
@@ -39,7 +39,9 @@ class SafeJobRunner(private val jobRepository: JobRepository, private val monito
       } else {
         log.warn("Ignoring job ${job.jobType} due to failure in a dependent job.")
 
-        jobRepository.saveAndFlush(Job.failed(job.jobType, LocalDateTime.now()))
+        jobRepository.saveAndFlush(Job.failed(job.jobType, LocalDateTime.now())).also { failedJob ->
+          monitoringService.capture("Job '${failedJob.jobType}' for job id '${failedJob.jobId}' failed")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/job/SafeJobRunner.kt:43]: Missed off previous PR.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/job/SafeJobRunner.kt [hunk 3]
```diff
@@ -39,7 +39,9 @@ class SafeJobRunner(private val jobRepository: JobRepository, private val monito
       } else {
         log.warn("Ignoring job ${job.jobType} due to failure in a dependent job.")
 
-        jobRepository.saveAndFlush(Job.failed(job.jobType, LocalDateTime.now()))
+        jobRepository.saveAndFlush(Job.failed(job.jobType, LocalDateTime.now())).also { failedJob ->
+          monitoringService.capture("Dependant job '${failedJob.jobType}' for job id '${failedJob.jobId}' failed")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/job/SafeJobRunner.kt:43]: Should capture here as well.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ManageAttendancesService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ManageAttendancesService.kt:87]: Would it continue after the failure? i.e. just affect one attendance row, or one activity? If they're all failing, not sure we want to be spammed with 1300 alerts?  The 5-mile-island problem!

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**